src/core/metadata/qgsabstractlayermetadataprovider.cpp

```diff
@@ -116,3 +117,25 @@ void QgsLayerMetadataProviderResult::setStandardUri( const QString &standardUri
 {
   mStandardUri = standardUri;
 }
+
+QIcon QgsLayerMetadataProviderResult::layerTypeIcon() const
```

Comment:
This looks identical to QgsIconUtils::iconForLayer(... ). Could you instead move the logic from QgsIconUtils::iconForLayer to a new QgsIconUtils::iconForLayer( QgsMapLayerType, QgsWkbType ) override, and call that from QgsIconUtils::iconForLayer ?

That'll allow removal of this duplicate logic/code.